apps/api/src/services/sendgrid_handler.py

```diff
@@ -0,0 +1,70 @@
+# using SendGrid's Python Library
+# https://github.com/sendgrid/sendgrid-python
+import os
+from logging import getLogger
+from typing import Iterable, Tuple, TypedDict, Union
+from typing_extensions import NotRequired
+
+import aiosendgrid
+from httpx import HTTPStatusError
+from sendgrid.helpers.mail import Email, Mail, Personalization
+
+log = getLogger(__name__)
+
+SENDGRID_API_KEY = os.getenv("SENDGRID_API_KEY")
+
+
+class PersonalizationData(TypedDict):
+    email: str
+    first_name: NotRequired[str]
+    last_name: NotRequired[str]
+    passphrase: NotRequired[str]
```

Comment:
Suggestion: Given how this is template specific data, I think we can move the `Template` enum from `email_handler` to here, make `PersonalizationData` contain just the `email`, make subtypes for each template, and then overload `send_email` to use a specific data type corresponding to the template.

Reply:
What exactly do you mean by "subtypes"?

Reply:
We can extend a base version of the personalization type as needed
```python
class PersonalizationData(TypedDict):
    email: str

class ConfirmationPersonalization(PersonalizationData):
    first_name: str
    last_name: str

class GuestTokenPersonalization(PersonalizationData):
    passphrase: str
```

Reply:
Added in 703c09e